husarion_ugv_lights/test/unit/led_components/test_led_segment.cpp

```diff
@@ -201,33 +196,18 @@ TEST_F(TestLEDSegment, SetAnimation)
   const auto charging_anim_desc = YAML::Load("{duration: 2}");
 
   EXPECT_NO_THROW(
-    led_segment_->SetAnimation("husarion_ugv_lights::ImageAnimation", image_anim_desc, false));
+    led_segment_->SetAnimation("husarion_ugv_lights::ImageAnimation", image_anim_desc, 0, false));
 
   EXPECT_NO_THROW(led_segment_->SetAnimation(
-    "husarion_ugv_lights::ChargingAnimation", charging_anim_desc, false, "0.5"));
-}
-
-TEST_F(TestLEDSegment, SetAnimationRepeating)
-{
-  const auto anim_desc = YAML::Load(
-    "{image: $(find husarion_ugv_lights)/test/files/animation.png, "
-    "duration: 2}");
-  ASSERT_NO_THROW(
-    led_segment_->SetAnimation("husarion_ugv_lights::ImageAnimation", anim_desc, false));
-
-  EXPECT_TRUE(led_segment_->GetDefaultAnimation().get() == nullptr);
-
-  ASSERT_NO_THROW(
-    led_segment_->SetAnimation("husarion_ugv_lights::ImageAnimation", anim_desc, true));
-
-  EXPECT_TRUE(led_segment_->GetDefaultAnimation().get() != nullptr);
-  EXPECT_TRUE(led_segment_->IsAnimationFinished());
+    "husarion_ugv_lights::ChargingAnimation", charging_anim_desc, 0, false, "0.5"));
 }
 
 TEST_F(TestLEDSegment, UpdateAnimationAnimationNotSet)
 {
-  EXPECT_TRUE(husarion_ugv_utils::test_utils::IsMessageThrown<std::runtime_error>(
-    [&]() { led_segment_->UpdateAnimation(); }, "Segment animation not defined"));
+  EXPECT_FALSE(husarion_ugv_utils::test_utils::IsMessageThrown<std::runtime_error>(
+    [&]() { led_segment_->UpdateAnimation(); },
+    "Segment animation not defined"));  // with the current implementation, this should not throw an
+                                        // error
 }
```

Comment:
Again, what does this test? What is expected behavior if we call UpdateAnimation when animation is not set?